hummingbot/connector/exchange_base.pyx

```diff
@@ -283,3 +281,43 @@ cdef class ExchangeBase(ConnectorBase):
         required volume.
         """
         return Decimal(str(self.get_price_for_volume(trading_pair, is_buy, amount).result_price))
+
+    async def get_required_collateral(
```

Comment:
If we make the function receive the tokens individually (base and quote) and also the fee as parameters, we could move it to the `PerpetualTrading` class.

Reply:
That was my original approach, but by placing the function in the `ExchangeBase`, we have a unified way for perpetual as well as spot exchanges to calculate the true size of a proposed position